### ISSUE TITLE
it is unnecessary to process config files which starting with .

### ISSUE THREAD

**Issue by 4admin2root** [kubernetes/kubernetes#44990]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.): no

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):
kubelet manifests
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):BUG REPORT

**Kubernetes version** (use `kubectl version`):1.6.1

**Environment**: 
- **Cloud provider or hardware configuration**: openstack
- **OS** (e.g. from /etc/os-release): centos 7.2
- **Kernel** (e.g. `uname -a`): 3.10.0-327.el7.x86_64
- **Install tools**: kubeadm
- **Others**:

**What happened**:
If I edit a file in /etc/kubernetes/manifests/
the kubelet will process config files which starting with .
exmple: /etc/kubernetes/manifests/.kube-controller-manager.yaml.swp

**What you expected to happen**:
it is unnecessary to process config files which starting with .

**How to reproduce it** (as minimally and precisely as possible):
vim /etc/kubernetes/manifests/kube-controller-manager.yaml

**Anything else we need to know**:
none

**Comment by Random-Liu** [comment]
This should have been fixed. Doesn't that work? https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/file_linux.go#L84

**Comment by Random-Liu** [comment]
It seems that when using `vim`, there are some other temporary files ended with `~`. Those files are not ignored. That is a known issue.

For now, it is recommended to edit the manifest outside that directory and copy the file back.

**Comment by resouer** [comment]
It should be already fixed. 

Some special files like prefix `~` is not common (my vim did not show that actually).

**Comment by antoineco** [comment]
It's not fixed (v1.6.4):
```
May 30 13:12:05 host-192-168-0-1 kubelet-wrapper[936]: E0530 13:12:05.404643     936 file_linux.go:113] can't process config file "/etc/kubernetes/manifests/.kube-addon-manager.yml.swp": /etc/kubernetes/manifests/.kube-addon-manager.yml.swp: read 'b0VIM 8.0
May 30 13:12:07 host-192-168-0-1 kubelet-wrapper[936]: E0530 13:12:07.731651     936 file.go:72] unable to read config path "/etc/kubernetes/manifests": error while processing event ("/etc/kubernetes/manifests/kube-addon-manager.yml": 0x40 == IN_MOVED_FROM): the pod with key kube-system/kube-addon-manager-host-192-168-0-1 doesn't exist in cache
```

**Comment by andyxning** [comment]
@antoineco In fact, this has not been fixed in v1.6.4. The filter for **files** starting with a dot(`.kube-addon-manager.yml.swp`) instead of *file paths starting with  a dot**(`./etc/kubernetes/manifests/.kube-addon-manager.yml.swp`), has been fixed in #45111 and only available after Kubernetes v1.7.0-alpha.4.

So we can close this now, imo.
